### PR TITLE
Docs for formatTreeList().

### PR DESCRIPTION
**Pull request by dereuromark**
Beginning of docs for https://github.com/cakephp/cakephp/pull/6704
Feel free to adjust

### CODE REVIEW

**Comment by markstory** [comment]
Looks good so far.

File: en/core-libraries/behaviors/tree.rst [hunk 1]
```diff
@@ -453,6 +453,24 @@ are a few more tree-orientated permutations at your disposal.
...
+    :ref:`model-find-list` but with a nested prefix that is specified
+    in the ``spacer`` option to show the structure of your data.
+    See generateTreeList() for an example output.
+
+    Supported options are:
+
+    * ``keyPath``: A string path to the key, i.e. "{n}.Post.id".
+    * ``valuePath``: A string path to the value, i.e. "{n}.Post.title".
+    * ``spacer``: The character or characters which will be repeated.
+
```
Comment (lorenzo) [inline en/core-libraries/behaviors/tree.rst]: Adding an example would be useful

Comment (dereuromark) [inline en/core-libraries/behaviors/tree.rst]: The example is a few lines above for generateTreeList(). Thus I thought duplicating that in such a close proximity doesn't do it much good.

Comment (lorenzo) [inline en/core-libraries/behaviors/tree.rst]: I think it is good to show how to use it in combination with results from another find

Comment (lorenzo) [inline en/core-libraries/behaviors/tree.rst]: it does not have to show the result, just an usage example

**Comment by dereuromark** [comment]
Anyone time to add an example and merge it?

**Comment by markstory** [comment]
@dereuromark I can take care of that.

**Pull request merged**